jerry-core/ecma/operations/ecma-objects.c

```diff
@@ -1322,10 +1326,19 @@ ecma_op_object_put_with_receiver (ecma_object_t *object_p, /**< the object */
     }
     else if (type == ECMA_OBJECT_TYPE_FUNCTION)
     {
+#if ENABLED (JERRY_ES2015)
+      /* Uninitialized 'length' property is non-writable (ECMA-262 v6, 19.2.4.1) */
+      if ((ecma_string_is_length (property_name_p))
+          && (!ECMA_GET_FIRST_BIT_FROM_POINTER_TAG (((ecma_extended_object_t *) object_p)->u.function.scope_cp)))
+      {
+        return ecma_reject (is_throw);
+      }
+#else /* !ENABLED (JERRY_ES2015) */
```

Comment:
Would it work if we put `#if !ENABLED (JERRY_ES2015)` here and remove the code above? The property would be instantiated (I suspect this is very rare case), and rejected because it is non-writable by default.

Reply:
Yes, It would work correctly even if the above code removed.
I put the check code here to avoid instantiation of 'length' property.